keras_cv/layers/preprocessing/rgb_shift.py

```diff
@@ -0,0 +1,144 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np 
+import tensorflow as tf
+from tensorflow.keras import layers, backend
+
+
+class RGBShift(layers.Layer):
+    """RGBShift class randomly shift values for each channel of the input RGB image. The expected images 
+    should be [0-255] pixel ranges.
+
+   
+    Input shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+    Output shape:
+        3D (unbatched) or 4D (batched) tensor with shape:
+        `(..., height, width, channels)`, in `"channels_last"` format
+
+    Args:
+        r_shift_limit: A scalar or tuple or list of two upper and lower bound number. It will change r-channel by
+            adding with the r-channel. If r_shift_limit is a single integer, the range will be (-r_shift_limit, r_shift_limit).
+            Default: (-10, 10).
+        g_shift_limit: A scalar or tuple or list of two upper and lower bound number. It will change g-channel by
+            adding with the r-channel. If r_shift_limit is a single integer, the range will be (-g_shift_limit, g_shift_limit).
+            Default: (-10, 10).
+        b_shift_limit: A scalar or tuple or list of two upper and lower bound number. It will change b-channel by
+            adding with the r-channel. If r_shift_limit is a single integer, the range will be (-b_shift_limit, b_shift_limit).
+            Default: (-10, 10).
+        seed: Integer. Used to create a random seed. Default: None.
+
+    Usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    rgbshift = keras_cv.layers.preprocessing.RGBShift()
+    augmented_images = rgbshift(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        r_shift_limit=10,
+        g_shift_limit=10,
+        b_shift_limit=10,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+        self._r_shift_limit = self._set_shift_limit(r_shift_limit, channel='r_shift_limit')
+        self._g_shift_limit = self._set_shift_limit(g_shift_limit, channel='g_shift_limit')
+        self._b_shift_limit = self._set_shift_limit(b_shift_limit, channel='b_shift_limit')
+        self.seed = seed
+
+    def _set_shift_limit(self, shift_limit, channel=''):
+        if isinstance(shift_limit, (tuple, list)):
+            if len(shift_limit) != 2: 
+                raise ValueError(f'The {channel} should be scalar, tuple or list of two upper and lower \
+                    bound number. Got {shift_limit}')
+            return sorted(shift_limit)
+        elif isinstance(shift_limit, (int, float)):
+            shift_limit = abs(shift_limit)
+            return [-shift_limit, shift_limit]
+        else:
+            raise ValueError(f'The {channel} should be scalar, tuple or list of two upper and lower bound\
+                 number. Got {shift_limit}')
+
+    @tf.function
```

Comment:
I don't think tf.function is necessary here, unless this is strictly needed.

Reply:
Done.